bpf/aya-bpf/src/programs/tracepoint.rs

```diff
@@ -0,0 +1,22 @@
+use crate::{helpers::bpf_probe_read, BpfContext};
+use core::ffi::c_void;
+
+pub struct TracePointContext {
+    ctx: *mut c_void,
+}
+
+impl TracePointContext {
+    pub fn new(ctx: *mut c_void) -> TracePointContext {
+        TracePointContext { ctx }
+    }
+
+    pub unsafe fn read_at<T>(&self, offset: usize) -> Result<T, i64> {
+        bpf_probe_read(self.ctx.add(offset) as *const T)
```

Comment:
I think you want to cast self.ctx to usize and add usizes here. Otherwise you're adding mem::size_of::<*mut c_void>() * count.

Reply:
I don't think so, according to the [doc of add](https://doc.rust-lang.org/std/primitive.pointer.html#method.add), we're adding `count * mem::size_of::<c_void>` here which is expected.

Reply:
oops yes of course you're right